erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+from pypika import functions as fn
+
+from erpnext.utilities.bulk_transaction import task, update_logger
+
+
+class BulkTransactionLog(Document):
+    pass
+
+
+@frappe.whitelist()
+def retry_failing_transaction():
+    btp = frappe.qb.DocType("Bulk Transaction Log Detail")
+    data = (
+        frappe.qb.from_(btp)
+        .select(btp.transaction_name, btp.from_doctype, btp.to_doctype)
+        .distinct()
+        .where(btp.retried != 1)
+        .where(btp.transaction_status == "Failed")
+        .where(btp.date == fn.CurDate())
+    ).run(as_dict=True)
+    if len(data) > 10:
+        frappe.enqueue(job, queue="long", job_name="bulk_retry", data=data)
+    else:
+        job(data)
+
+
+def job(data):
+    for d in data:
+        failed = []
+        try:
+            task(d.transaction_name, d.from_doctype, d.to_doctype)
+        except Exception as e:
```

Comment:
this will commit partial data, use savepoints/rollback like the other function which is also named job. Also possibly duplicate code?

Reply:
Added save points and rollback